TweetMate/src/main/java/com/ybsystem/tweetmate/usecases/ClickUseCase.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    public static void openURL(String url) {
+        // Intent to Browser
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+        TweetMateApp.getActivity().startActivity(intent);
+    }
+
+    public static void searchWord(String searchWord) {
+        // Intent to SearchActivity
+        Activity act = TweetMateApp.getActivity();
+        Intent intent = new Intent(act, SearchActivity.class);
+        intent.putExtra("SEARCH_WORD", searchWord);
+        act.startActivityForResult(intent, 0);
+        act.overridePendingTransition(R.anim.slide_in_from_right, R.anim.zoom_out);
+    }
+
+    public static void tweetWithPrefix(String word) {
+        // Intent to PostActivity
+        Activity act = TweetMateApp.getActivity();
+        Intent intent = new Intent(act, PostActivity.class);
+        intent.putExtra("TWEET_PREFIX", word);
+        act.startActivity(intent);
+        act.overridePendingTransition(R.anim.fade_in_from_bottom, R.anim.none);
+    }
+
+    public static void tweetWithSuffix(String word) {
+        // Intent to PostActivity
+        Activity act = TweetMateApp.getActivity();
+        Intent intent = new Intent(act, PostActivity.class);
+        intent.putExtra("TWEET_SUFFIX", word);
+        act.startActivity(intent);
+        act.overridePendingTransition(R.anim.fade_in_from_bottom, R.anim.none);
+    }
+
+    public static void showUser(long userId) {
+        // Intent to ProfileActivity
+        Activity act = TweetMateApp.getActivity();
+        Intent intent = new Intent(act, ProfileActivity.class);
+        intent.putExtra("USER_ID", userId);
+        act.startActivity(intent);
+        act.overridePendingTransition(R.anim.slide_in_from_right, R.anim.zoom_out);
+    }
+
+    public static void showDetail(TwitterStatus status) {
+        // Intent to TimelineActivity
+        Activity act = TweetMateApp.getActivity();
+        Intent intent = new Intent(act, TimelineActivity.class);
+        intent.putExtra("STATUS", status);
+        intent.putExtra("COLUMN_TYPE", DETAIL);
+        act.startActivity(intent);
+        act.overridePendingTransition(R.anim.slide_in_from_right, R.anim.zoom_out);
+    }
+
+    public static void copyText(TwitterStatus status) {
+        // Get clipboard
+        ClipboardManager clipboardManager = (ClipboardManager)
+                TweetMateApp.getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+        // Check null
+        if (clipboardManager != null) {
+            clipboardManager.setPrimaryClip(ClipData.newPlainText("", status.getText()));
+            ToastUtils.showShortToast(STR_SUCCESS_COPY_TWEET);
+        }
+    }
+
+    public static void share(TwitterStatus status) {
+        // Create uri
+        String screenName = status.getUser().getScreenName();
+        String statusId = String.valueOf(status.getId());
+        Uri uri = Uri.parse("https://twitter.com/" + screenName + "/status/" + statusId);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "https://twitter.com/" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1192)